pkg/render/apiserver.go

```diff
@@ -290,6 +293,47 @@ func (c *apiServerComponent) Ready() bool {
 	return true
 }
 
+// For legacy reasons we use apiserver: true here instead of the k8s-app: name label,
+// so we need to set it explicitly rather than use the common labeling logic.
+func (c *apiServerComponent) deploymentSelector() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"apiserver": "true",
+		},
+	}
+}
+
+// Determine names based on the configured variant
+// It takes two name as parameters, enterpriseName and ossName, and returns name and nameToDelete.
+func (c *apiServerComponent) resourceNameBasedOnVariant(enterpriseName, ossName string) (string, string) {
+	var name, nameToDelete string
+	switch c.cfg.Installation.Variant {
+	case operatorv1.TigeraSecureEnterprise:
+		name = enterpriseName
+		nameToDelete = ossName
+	case operatorv1.Calico:
+		name = ossName
+		nameToDelete = enterpriseName
+	}
+	return name, nameToDelete
+}
+
+func (c *apiServerComponent) apiServerPodDisruptionBudget() *policyv1.PodDisruptionBudget {
+	maxUnavailable := intstr.FromInt(1)
+	name, _ := c.resourceNameBasedOnVariant("tigera-apiserver", "calico-apiserver")
+	return &policyv1.PodDisruptionBudget{
+		TypeMeta: metav1.TypeMeta{Kind: "PodDisruptionBudget", APIVersion: "policy/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: rmeta.APIServerNamespace(c.cfg.Installation.Variant),
```

Comment:
This use to be `Namespace: common.CalicoNamespace`, but I think it needs to depend on the variant.

Reply:
Thanks. Had issues with accessing the CI error log or generating locally.